### ISSUE TITLE
Wheather the so-called attebtion distillation machanism work?

### ISSUE THREAD

**Issue by SmILeel** [bboylyg/NAD#17]
Hi, dear author.

Based on the codes you provided, i have done some experiments by changing the hyper-parameters.

However, When I mask the feature attention loss in training, and only use the cls_loss as follows, it still could get a good results.
\# in train_step function
  cls_loss = criterionCls(output_s, target)
  \# at3_loss = criterionAT(activation3_s, activation3_t.detach()) * opt.beta3
  \# at2_loss = criterionAT(activation2_s, activation2_t.detach()) * opt.beta2
  \# at1_loss = criterionAT(activation1_s, activation1_t.detach()) * opt.beta1
  \# at_loss = at1_loss + at2_loss + at3_loss + cls_loss
  at_loss = cls_loss

The results are:
epoch: 19  lr: 0.0100
testing the models......
[clean]Prec@1: 83.41
[bad]Prec@1: 6.18

I wonder whether the attention distillation works in the processes, because just retraininig on small clean set could remove  the backdoor in your checkpoints.

Besides, when i am applying your meshnism in other networks, It dosen't work no matter how i change the hyperparameters.

**Comment by bboylyg** [comment]
Hi,
The issue stems from the absence of standard data augmentation, such as flip and random crop, in the [train_badnet.py](https://github.com/bboylyg/NAD/blob/main/train_badnet.py)) script used to train the backdoored model. As a result, even basic fine-tuning or retraining (without our distillation loss) can significantly lower the attack success rate. To address this, we recommend training the backdoored model using standard data augmentation techniques (such as flip and random crop) before reassessing the effectiveness of the distillation loss.

By the way, you might want to take a look at the other closed issues. They could provide some valuable insights on how NAD's distillation loss works.

**Comment by SmILeel** [comment]
Thanks for your reply, and i would check the codes with more experiments and read the closed issues.  Anyway,  thanks for your sharing.

**Comment by SmILeel** [comment]
By the way,  there is an open sourced leaderboard for backdoor defense, and the NAD even performance worse than just finetune the model. This is similar to the results i get on my experiments.
http://[backdoorbench.com/leader_cifar10](http://backdoorbench.com/leader_cifar10)

**Comment by SmILeel** [comment]
http://[backdoorbench.com/leader_cifar10](http://backdoorbench.com/leader_cifar10)
By the way, there is an leaderboad for backdoor defense and you can see that the NAD even performance worse than just finetuning, which means the atttion machanism doesn't work at all. And this is similar to what i have seen on my experiments.
![image](https://github.com/bboylyg/NAD/assets/107383023/d7b648c2-87be-4d5a-9228-25f0c8aa52c5)

**Comment by SmILeel** [comment]
![image](https://github.com/bboylyg/NAD/assets/107383023/1dcc09df-423e-471b-b15d-68ff439e6abf)
![image](https://github.com/bboylyg/NAD/assets/107383023/1d2deb38-16e5-4018-bc86-301936ca3aa0)
![image](https://github.com/bboylyg/NAD/assets/107383023/c3590e03-ca4c-4f8f-9ed3-95b121ed3402)
There and more experiments for different models, and they get the same conclusion.

**Comment by bboylyg** [comment]
Please do comprehensive experiments by yourself and do not blindly rely on the results without further verification, which is not fairly to our work. One more thing I would let you know is that increasing distillation hyperparameter $\beta$ can significantly improve the defense performance you can have a try.  

Please take note that the efficiency of NAD's distillation loss, which employs flexible $\beta$ searching, has been extensively assessed in relevant papers, including [DBD](https://arxiv.org/pdf/2202.03423.pdf) and our latest publication, [RNP](https://arxiv.org/pdf/2305.14876.pdf). I highly recommend that you take the time to review them.

**Comment by SmILeel** [comment]
Thanks for reply, i would do more further check.
